services/121-service/src/registration-events/repositories/registration-event.view.repository.ts

```diff
@@ -0,0 +1,39 @@
+import { Inject } from '@nestjs/common';
+import { REQUEST } from '@nestjs/core';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+
+import {
+  RegistrationEventViewEntity,
+  STATUS_CHANGE_STRING,
+} from '@121-service/src/registration-events/entities/registration-event.view.entity';
+import { ScopedRepository } from '@121-service/src/scoped.repository';
+import { ScopedUserRequest } from '@121-service/src/shared/scoped-user-request';
+
+export class RegistrationEventViewScopedRepository extends ScopedRepository<RegistrationEventViewEntity> {
+  constructor(
+    @Inject(REQUEST) request: ScopedUserRequest,
+    @InjectRepository(RegistrationEventViewEntity)
+    repository: Repository<RegistrationEventViewEntity>,
+  ) {
+    super(request, repository);
+  }
+
+  // ##TODO: can this take over some or all of normal repository methods as well?
```

Comment:
- For inserts and saves you cannot use the views
- For other things (like find) I would say yes we can use this, if it more convienent as than the joins are abstracted in the view